### DISCUSSION TITLE
path.filename_first on all results with pahts

### DISCUSSION THREAD

**Discussion by nela** [ibhagwan/fzf-lua#1511] [Q&A] [answered]
Firstly, thanks for the great plugin - it is at the centre of my workflow! 

I've newly found the option `{ files = { formatter = "path.filename_first" }  }` which is quite lovely and useful. So nuch, that it would be nice to include it on all queries that output filenames. `grep/live_grep` and lsp `location_handler` are the first that come to mind where this could be extended. 

Moreover, I wonder if there is any existing functionality that allows for truncating long filenames (regardless of `path.filename_first` formatting). That a long filename such as `src/app/common/something/else.sh` could be truncated to `s/a/c/s/else.sh`, also toggling between these options would be great.

If not, I'd be happy to help with developing these, albeit some guideance would be very appreciated.

**Comment by ibhagwan** [comment, marked as answer]
> that it would be nice to include it on all queries that output filenames

Add it to defaults (I would also recommend to try v2 which helps with logical search, see #1255):
```lua
:lua require("fzf-lua").setup({ defaults = { formatter={"path.filename_first",2} } })
```

> I wonder if there is any existing functionality that allows for truncating long filenames

Use `path_shorten=true` (can be combined with the formatter above).

**Reply by nela** [reply]
Perfect, thanks! 
Is it possible to make `path_shorten=true` work with grep/live_grep?
Also, is there any way I can pass these formatting options to `make_entry.file`  or `opts` in `fzf_exec` ? I've tried setting both, but haven't been able to make them work.

**Reply by ibhagwan** [reply]
> Is it possible to make path_shorten=true work with grep/live_grep?

It works with all pickers displaying file entries, add it to defaults if you want it for everything.

> Also, is there any way I can pass these formatting options to make_entry.file or opts in fzf_exec ? I've tried setting both, but haven't been able to make them work.

Pass the opts as 2nd arg to `make_entry.file`:

https://github.com/ibhagwan/fzf-lua/blob/ce1e24f2a48089b65deefcb0a71a9856839002b5/lua/fzf-lua/make_entry.lua#L305

**Reply by nela** [reply]
Thanks!
